Cargo.lock

```diff
@@ -4787,12 +4907,50 @@ dependencies = [
  "url",
 ]
 
+[[package]]
+name = "openssl"
+version = "0.10.64"
+source = "registry+https://github.com/rust-lang/crates.io-index"
+checksum = "95a0481286a310808298130d22dd1fef0fa571e05a8f44ec801801e84b216b1f"
+dependencies = [
+ "bitflags 2.5.0",
+ "cfg-if",
+ "foreign-types",
+ "libc",
+ "once_cell",
+ "openssl-macros",
+ "openssl-sys",
+]
```

Comment:
Are we introducing a dependency on a system openssl? It's probably ok but maybe we should document it. (@pimeys )

Reply:
hum good point, didn't notice. Only used in tests for now so should be a problem, I guess?